File: hr-hourglass_sum-arrays.cpp
// Complete the hourglassSum function below.
int hourglassSum(vector<vector<int>> arr) { // O(N^2)
    int rows = arr.size();
    int cols = arr[0].size();
    int maximum_hourglass = -INFINITY;
    int current_hourglass = 0;
    int i, j;
    for (i = 1; i < rows-1; ++i) {
        for (j = 1; j < cols-1; ++j) {
            current_hourglass = arr[i][j]
                + arr[i-1][j-1] + arr[i-1][j] + arr[i-1][j+1]
                + arr[i+1][j-1] + arr[i+1][j] + arr[i+1][j+1];
            if (current_hourglass > maximum_hourglass) {
                maximum_hourglass = current_hourglass;
            }
        }
    }
    return maximum_hourglass;
}

File: hr-hourglass_sum-arrays.py
# Complete the hourglassSum function below.
def hourglassSum(arr):
    maximum_hourglass = -math.inf
    current_hourglass = 0
    rows = len(arr)
    cols = len(arr[0])
    for i in range(1,rows-1):
        for j in range(1,cols-1):
            current_hourglass = arr[i][j] + arr[i-1][j-1] + arr[i-1][j] + arr[i-1][j+1]                     + arr[i+1][j-1] + arr[i+1][j] + arr[i+1][j+1]
            if current_hourglass > maximum_hourglass:
                maximum_hourglass = current_hourglass
    return maximum_hourglass
